packages/firebase_ml_custom/example/lib/main.dart

```diff
@@ -29,61 +29,98 @@ class _MyAppState extends State<MyApp> {
     var modelFile = await loadModelFromFirebase();
     return await loadTFLiteModel(modelFile);
   }
 
+  // Download custom model from the Firebase console and return its file
+  // located on the mobile device
   static Future<File> loadModelFromFirebase() async {
-    var model = FirebaseCustomRemoteModel('mobilenet_v1_1_0_224');
-
-    var conditions = FirebaseModelDownloadConditions(androidRequireWifi: true);
-    var modelManager = FirebaseModelManager.instance;
-
-    await modelManager.download(model, conditions);
-    assert(await modelManager.isModelDownloaded(model) == true);
-
-    var modelFile = await modelManager.getLatestModelFile(model);
-    assert(modelFile != null);
-    return modelFile;
+    try {
+      // Create model with a name that is specified in the Firebase console
+      var model = FirebaseCustomRemoteModel('mobilenet_v1_1_0_224');
+
+      // Specify conditions when the model can be downloaded.
+      // If there is no wifi access when the app is started,
+      // this app will continue loading until the conditions are satisfied.
+      var conditions = FirebaseModelDownloadConditions(
+          androidRequireWifi: true, iosAllowCellularAccess: false);
+
+      // Create model manager associated with default Firebase App instance.
+      var modelManager = FirebaseModelManager.instance;
+
+      // Begin downloading and wait until the model is downloaded successfully.
+      await modelManager.download(model, conditions);
+      assert(await modelManager.isModelDownloaded(model) == true);
+
+      // Get latest model file to use it for inference by the interpreter.
+      var modelFile = await modelManager.getLatestModelFile(model);
+      assert(modelFile != null);
+      return modelFile;
+    }
+    catch (exception) {
+      print('Failed on loading your model from Firebase: $exception');
+      print('The program will not be resumed');
+      rethrow;
+    }
   }
 
+  // Load the model into some TF Lite interpreter.
+  // In this case interpreter provided by tflite plugin
   static Future<String> loadTFLiteModel(File modelFile) async {
-    var appDirectory = await getApplicationDocumentsDirectory();
-    var labelsData =
-        await rootBundle.load("assets/labels_mobilenet_v1_224.txt");
-    var labelsFile =
-        await File(appDirectory.path + "_labels_mobilenet_v1_224.txt")
-            .writeAsBytes(labelsData.buffer.asUint8List(
-                labelsData.offsetInBytes, labelsData.lengthInBytes));
-
-    assert(await Tflite.loadModel(
-          model: modelFile.path,
-          labels: labelsFile.path,
-          isAsset: false,
-        ) ==
-        "success");
-    return "Model is loaded";
+    try{
+      var appDirectory = await getApplicationDocumentsDirectory();
+      var labelsData =
+          await rootBundle.load("assets/labels_mobilenet_v1_224.txt");
+      var labelsFile =
+          await File(appDirectory.path + "_labels_mobilenet_v1_224.txt")
```

Comment:
nit: will it break things to add a "/"?

This example appends a "/":
https://flutter.dev/docs/cookbook/persistence/reading-writing-files

Reply:
I know for fact it doesn't break linux, seems like my IOS and Android devices aren't broken either. So, I'm adding it to the code.